packages/framework/ai-collab/src/aiCollabUiDiffApi.ts

```diff
@@ -0,0 +1,200 @@
+
+/**
+ * A base interface to enforce consistency between all Diff objects.
+ * @remarks This object is not intended to be used directly.
+ * The union type Diff provides a better typescript experience
+ *
+ * @alpha
+ */
+export interface DiffBase {
+	/**
+	 * The operation type performed by an ai agent on a SharedTree
+	 * @remarks This is intended to be used to correlate the diff with the operation that generated it.
+	 */
+	readonly type: string;
+	/**
+	 * An explanation from the ai as to why the edit is being made.
+	 */
+	readonly aiExplanation: string;
+}
+
+/**
+ * An object that provides relevant information to visualize a single edit performed by an ai agent on a SharedTree
+ * @alpha
+ */
+export type Diff = InsertDiff | ModifyDiff | RemoveDiff | MoveDiff;
+
+/**
+ * A path from the root of the tree node passed to ai-collab to a specific node within the tree.
+ * @alpha
+ */
+export type NodePath = {
+	/**
+	 * The short id of the node.
+	 * @remarks the root tree node and nodes without a defined SchemaFactory.identifier field will not have a short id.
+	 */
+	shortId: string | number | undefined;
+	/**
+	 * The schema of the node.
+	 */
+	schemaIdentifier: string;
+	/**
+	 * The field within the parent node that the node is located at.
+	 * @remarks
+	 * The root node will have a parentField name of 'rootFieldKey'.
+	 * Nodes in an array use numbers to represent their index within the array.
+	 */
+	parentField: string | number;
+}[];
+
+/**
+ * An object that describes the insertion of a new node into a tree.
+ * @remarks This object is intended to be used to visualize the changes made to a tree on a UI.
+ * @alpha
+ */
+export interface InsertDiff extends DiffBase {
+	type: "insert";
+	/**
+	 * The path from the root node to the newly inserted node.
+	 * The last value in the path will be the newly inserted node.
+	 * If the newly inserted node is a primitive value, the last value in the path will be the parent array node.
+	 */
+	nodePath: NodePath;
+	/**
+	 * The content of the newly inserted node.
+	 */
+	nodeContent: unknown;
+}
+
+/**
+ * An object that describes the modification of an existing node on a tree.
+ * @remarks This object is intended to be used to visualize the changes made to a tree on a UI.
+ * @alpha
+ */
+export interface ModifyDiff extends DiffBase {
+	type: "modify";
+	/**
+	 * The path from the root node to the ndoe being modified.
+	 */
+	nodePath: NodePath;
+	/**
+	 * The new value of the node.
+	 */
+	newValue: unknown;
+	/**
+	 * The old value of the node.
+	 */
+	oldValue: unknown;
+}
+
+/**
+ * An object that describes the removal of one or more nodes from a tree.
+ * @alpha
+ */
+export type RemoveDiff = RemoveNodeDiff | ArraySingleRemoveDiff | ArrayRangeRemoveDiff;
+
+/**
+ * An object that describes a field being removed from a SharedTree.
+ * @alpha
+ */
+export interface RemoveNodeDiff extends DiffBase {
+	type: "remove";
+	removalType: "remove-field";
```

Comment:
We should document what this is. As these types are currently implemented, you would need to document it on each removal type. An alternative strategy would be to introduce a `RemovalDiffBase` interface and have all of the removal interfaces implement that. E.g.
```
export interface RemovalDiffBase extends DiffBase {
    readonly type: "remove";
    readonly removalType: string;
    ...
}

export interface RemoveNodeDiff extends RemovalDiffBase {
    readonly removalType: "remove-field";
    ...
}
```